tests/org.pitest.pitclipse.ui.tests/src/org/pitest/pitclipse/ui/behaviours/pageobjects/RunConfigurationSelector.java

```diff
@@ -59,25 +73,245 @@ public PitRunConfiguration getConfiguration(String configName) {
+            bot.textWithLabel(PitArgumentsTab.TEST_CLASS_TEXT).setText(config.getTestObject());
+        } else {
+            bot.radio(PitArgumentsTab.TEST_DIR_RADIO_TEXT).click();
+            bot.textWithLabel(PitArgumentsTab.TEST_DIR_TEXT).setText(config.getTestObject());
+
+        }
+        if (config.isRunInParallel()) {
+            bot.checkBox(PitPreferences.RUN_IN_PARALLEL_LABEL).select();
+        } else {
+            bot.checkBox(PitPreferences.RUN_IN_PARALLEL_LABEL).deselect();
+        }
+        if (config.isIncrementalAnalysis()) {
+            bot.checkBox(PitPreferences.INCREMENTAL_ANALYSIS_LABEL).select();
+        } else {
+            bot.checkBox(PitPreferences.INCREMENTAL_ANALYSIS_LABEL).deselect();
+        }
+        bot.textWithLabel(PitPreferences.EXCLUDED_CLASSES_LABEL).setText(config.getExcludedClasses());
+        bot.textWithLabel(PitPreferences.EXCLUDED_METHODS_LABEL).setText(config.getExcludedMethods());
+        bot.textWithLabel(PitPreferences.AVOID_CALLS_LABEL).setText(config.getAvoidCallsTo());
+        // close shell and save
+        closeConfigurationShell();
+    }
+
+    /**
+     * @param config where the projects are listed
+     * @return String which lists all projects of the configuration separated with
+     *         commas
+     */
+    private String getProjectsAsString(PitRunConfiguration config) {
+        Iterator<String> it = config.getProjects().iterator();
+        StringBuilder sb = new StringBuilder();
+        while (it.hasNext()) {
+            sb.append(it.next());
+            if (it.hasNext()) {
+                sb.append(',');
+            }
+        }
+        return sb.toString();
+    }
+
+    public void setMutatorGroup(String configurationName, Mutators mutatorGroup) {
+        activateMutatorsTab(configurationName);
+        bot.radio(mutatorGroup.getDescriptor()).click();
+        closeConfigurationShell();
+    }
+
+    public void setAdditionalCustomMutator(String configurationName, Mutators mutator) {
+        activateMutatorsTab(configurationName);
+        SWTBotRadio radioButton = bot.radio(PitMutatorsTab.CUSTOM_MUTATOR_RADIO_TEXT);
+        if (!radioButton.isSelected()) {
+            radioButton.click();
+        }
+        SWTBotTable table = bot.table();
+        setMutator(table, mutator);
+        closeConfigurationShell();
+    }
+
+    public void setOneCustomMutator(String configurationName, Mutators mutator) {
+        activateMutatorsTab(configurationName);
+        bot.radio(PitMutatorsTab.CUSTOM_MUTATOR_RADIO_TEXT).click();
+        uncheckAllMutators();
+        SWTBotTable table = bot.table();
+        setMutator(table, mutator);
+        closeConfigurationShell();
+    }
+
+    public void checkAllMutators(String configurationName) {
+        activateMutatorsTab(configurationName);
+        bot.radio(PitMutatorsTab.CUSTOM_MUTATOR_RADIO_TEXT).click();
+        SWTBotTable table = bot.table();
+        Display.getDefault().syncExec(() -> {
```

Comment:
Just for curiosity, SWTBot does not provide any API here to perform selection without accessing the underlying widget?

Reply:
I didn't find a way to get the number of items inside the table without the widget.

Reply:
OK, no problem :)